docs/NEW_GUIDE_DRAFT.en.md

```diff
@@ -0,0 +1,68 @@
+`NEAR CLI Core commands`
+login
+create-account
+    - implicit
+    - subaccount
+<!-- account Id is entered in this step -->
+manage-account
+    - keys
+        - view
+        - add
+        - delete
+        - nonce
+    - local-keys
+        <!-- Needs to be designed (with multikey management) -->
+    - contract
+        - deploy
+        - dev-deploy
+        - state
+        - wasm-hash
```

Comment:
Ok, but let's stick to `functions`, not `methods`.
https://docs.near.org/docs/api/rpc/contracts#call-a-contract-function
We are using `function` word in our docs.

Reply:
The docs use both...
![image](https://user-images.githubusercontent.com/304265/140118300-f9ba0db7-6719-45fb-867b-1cb457eca21c.png)

Yes, I find it misleading.

@thisisjoshford @amgando What is your opinion on "functions" vs "methods"?

Reply:
cc @BenKurrek

Reply:
Yea, this should really be more consistent... @bucanero and I are going to go through and restructure the docs so we can fix this. Now the question is should they be called functions or methods. To this, I say that in general, functions are "defined outside of classes while methods are defined inside of and part of classes" as per [this post](https://stackoverflow.com/questions/155609/whats-the-difference-between-a-method-and-a-function#:~:text=A%20very%20general%20definition%20of,of%20and%20part%20of%20classes.)

Read [this](https://www.codecademy.com/articles/fwd-js-methods-functions) also, it's a great article. I think for smart contract programming, it would make more sense to call them methods since they are associated with the smart contract class / structure / object - however you wanna define that.

Reply:
In `Rust` it's a bit more tricky: https://doc.rust-lang.org/rust-by-example/fn/methods.html
But in our case contract functions are `methods`.  [Example]( https://github.com/near-examples/rust-status-message/blob/master/src/lib.rs#L23).
In `AS` they are `functions`  because they are not associated with any class. [Example](https://github.com/near-examples/guest-book/blob/master/assembly/main.ts#L13).
But since our main language is `Rust`, let's stick to `methods`.
FYI: we can support more languages in the future, which means that there is no right answer to this question. Let's just be consistent.

Reply:
yea totally agree.

Reply:
> In `Rust` it's a bit more tricky: [doc.rust-lang.org/rust-by-example/fn/methods.html](https://doc.rust-lang.org/rust-by-example/fn/methods.html) But in our case contract functions are `methods`. [Example](https://github.com/near-examples/rust-status-message/blob/master/src/lib.rs?rgh-link-date=2021-11-04T13%3A21%3A48Z#L23). In `AS` they are `functions` because they are not associated with any class. [Example](https://github.com/near-examples/guest-book/blob/master/assembly/main.ts?rgh-link-date=2021-11-04T13%3A21%3A48Z#L13). But since our main language is `Rust`, let's stick to `methods`. FYI: we can support more languages in the future, which means that there is no right answer to this question. Let's just be consistent.

Well, it is methods based on the code that is written in our opinionated framework, but the code generated and what is actually called is actually functions.

The example given here is expanded to:
```
#[cfg(target_arch = "wasm32")]
#[no_mangle]
pub extern "C" fn set_status() {
    near_sdk::env::setup_panic_hook();
   // .. I removed the serde garbage
    let Input { message }: Input = near_sdk::serde_json::from_slice(
        &near_sdk::env::input().expect("Expected input since method has arguments."),
    )
    .expect("Failed to deserialize input from JSON.");
    let mut contract: StatusMessage = near_sdk::env::state_read().unwrap_or_default();
    contract.set_status(message);
    near_sdk::env::state_write(&contract);
}
```